### ISSUE TITLE
custom server url with git protocol broken?

### ISSUE THREAD

**Issue by davidszotten** [nedbat/watchgha#22]
Hi,

thanks for a cool tool!

i'm trying to use it with a custom github url that's using the git protocol, but it's not working, i think due to the slash after the server url at https://github.com/nedbat/watchgha/blob/22156e1fb5e7f27d17abfd4a1982e282c130b609/src/watchgha/watch_runs.py#L136

```
rf"(?:{re.escape(server_url)}/|git@github.com:)
                             ^
                           here
```

did you intentionally not add a test suite to this lib? i guess most of it is about side-effects which are hard to test, though it might be handy for the data finagling bits like extracting owner / repo name from repo urls + github server

**Comment by nedbat** [comment]
Can you show the URL you are using? (I didn't know you could have custom GitHub urls?) 

About the non-test-suite: yes, mostly this code is hard to test, but also I just slapped this together and it grew... :(

**Comment by davidszotten** [comment]
sorry i should have said `GITHUB_SERVER_URL`

in my case `git@git.mydomain.com` (repo url is `git@git.mydomain.com:username/repo`)

**Comment by nedbat** [comment]
Fixed in commit a87b77f (and a start on a test suite in 466eccb).

**Comment by nedbat** [comment]
This is now released as part of [watchgha 2.3.2](https://pypi.org/project/watchgha/2.3.2).

**Comment by davidszotten** [comment]
thanks! ✨
